eclipse-pass/pass-support
# Update nihms pmc deposits using email integration

This PR adds an email integration to read `Bulk submission` emails from nihms.  

The PR contains the following:
- Add Deposit.statusMessage attribute in pass-data-client model
- Set Deposit.depositStatusRef to `nihms-package:<nihms_package_name>` when the submission deposit is sent to nihms SFTP server.
- Read the nihms emails every 10 minutes and use the contents of the emails to update the submission's deposit status.  A success email will result in the deposit status being set to `ACCEPTED` and the deposit.depositStatusRef being set to `nihms-id:<manuscript_id from success email>`.  A failure email will result in the deposit status being set to `REJECTED` and the deposit.statusMessage set to failure message in the email.  The Deposit is updated in PASS.

I have a couple manual tests I want to perform:

- Run the email integration locally and attach to actual inbox of `pass-nihms@jh.edu` to read email and verify it comes in as expected
   - This has a pending action to determine if this is possible with the JHU inbox. I will test with a gmail account to move this along.
- Run integration test in stage with actual nihms deposits

I will be doing these tests over the next couple days, but the code should be stable enough to review now.
